android-json-form-wizard/src/main/java/com/vijay/jsonwizard/fragments/JsonWizardFormFragment.java

```diff
@@ -401,12 +442,32 @@ public void onClick(View view) {
                     assert getFragmentManager() != null;
                     presenter.checkAndStopCountdownAlarm();
                     getJsonApi().setPreviousPressed(true);
-                    getFragmentManager().popBackStack();
+                    getJsonApi().getStack().pop();
+                    getFragmentManager().popBackStack(getJsonApi().getStack().pop(), 0);
                 }
             }
         }
     }
 
+    private boolean nextStepHasNoSkipLogic() {
```

Comment:
The implementation of this method should be calling `stepHasNoSkipLogic(@Nullable String step)` with `getJsonApi().nextStep()` as the argument